server/utils/interfaces/authRequestInterface.ts

```diff
@@ -0,0 +1,9 @@
+import { Request } from 'express';
+
+export interface AuthRequestInterface extends Request {
+  userData: {
+    id: number,
+    name: string,
+    role: string,
+  }
+}
```

Comment:
Not sure about the nesting or extending Request
you wanna `type LoginBody = { name: string .... }` i think which is better, but you might not need this typing if you can very smartly reuse the typing that came out from `const body = await validator()` body should be typed somehow

Reply:
Here, I'm defining this interface to be able to put the user's data that came out of the token in the request object and pass it to the next middleware. I defined another type for the user data object only. If my way is wrong, I need to ask you to explain a lil bit more 😄